icicle/appUtils/ntt/tests/verification.cu

```diff
@@ -141,22 +198,35 @@ int main(int argc, char** argv)
   int count = INPLACE ? 1 : 10;
   if (INPLACE) {
     CHK_IF_RETURN(
-      cudaMemcpy(GpuOutputNew, GpuScalars, NTT_SIZE * BATCH_SIZE * sizeof(test_data), cudaMemcpyDeviceToDevice));
+      cudaMemcpy(GpuOutputNew, COLUMNS_BATCH? GpuScalarsTransposed : GpuScalars, NTT_SIZE * BATCH_SIZE * sizeof(test_data), cudaMemcpyDeviceToDevice));
   }
   CHK_IF_RETURN(benchmark(true /*=print*/, count));
 
+  if (COLUMNS_BATCH) {
+    transpose_batch<<<(NTT_SIZE * BATCH_SIZE + 256 - 1)/256,256>>>(GpuOutputNew, GpuScalarsTransposed, BATCH_SIZE, NTT_SIZE);
+    cudaDeviceSynchronize();
+    printf("trans err %d\n", cudaGetLastError());
+    CHK_IF_RETURN(
+      cudaMemcpy(GpuOutputNew, GpuScalarsTransposed, NTT_SIZE * BATCH_SIZE * sizeof(test_data), cudaMemcpyDeviceToDevice));
+    cudaDeviceSynchronize();
+    printf("copy err %d\n", cudaGetLastError());
```

Comment:
Are these `cudaDeviceSynchronize` needed or for debug?

Reply:
removed